types/codec.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+package cosmos_sdk;
+option go_package = "types";
+
+import "gogoproto/gogo.proto";
+
+message Coin {
+    option (gogoproto.stringer) = false;
+    string denom = 1;
+    string amount = 2 [(gogoproto.customtype) = "Int", (gogoproto.nullable) = false];
+}
+
+message StdTx {
+    repeated bytes msg = 1;
```

Comment:
Leaving `msg` as raw `bytes` allows for two things:

1. we can use the raw bytes form as serialized for signing purposes if we determine we actually don't need determinism
2. this allows us to reuse `StdTx` without making each app implement its own `Tx`. I still think it should be possible for apps to implement a custom `Tx` and `Tx` should likely just be an interface that `StdTx` implements. This likely requires some discussion.